ratpack-core/src/main/java/ratpack/http/client/RequestSpec.java

```diff
@@ -40,6 +42,23 @@
   RequestSpec redirects(int maxRedirects);
 
   /**
+   * Sets the {@link SSLContext} used for client and server SSL authentication.
+   *
+   * @param sslContext SSL context with keystore as well as trust store
+   * @return the {@link RequestSpec}
+   */
+  RequestSpec sslContext(SSLContext sslContext);
+
+  /**
+   * Factory method to create {@link SSLContext} used for client and server SSL authentication.
+   *
+   * @param factory provides a factory that will create {@link SSLContext} instance
+   * @return the {@link RequestSpec}
+   * @throws Exception this can be thrown from the action
+   */
+  RequestSpec sslContext(Factory<SSLContext> factory) throws Exception;
```

Comment:
I'm not sure this is warranted, since we just invoke the factory immediately (i.e. a user can do this).

If you think it is needed, it should be implemented as an interface default method.